graph/schema/devicedefinition.graphqls

```diff
@@ -0,0 +1,166 @@
+# DeviceDefinition related types
+
+extend type Query {
+  """
+  View a particular device definition.
+  """
+  devicedefinition(
+    """
+    criteria to search for a device definition
+    """
+    by: DevicedefinitionBy!
+  ): DeviceDefinition!
+
+  """
+  List device definitions.
+  """
+  devicedefinitions(
+    """
+    The number of device definition to retrieve.
+    Mutually exclusive with `last`.
+    """
+    first: Int
+    """
+    A cursor for pagination. Retrieve device definitions after this cursor.
+    """
+    after: String
+    """
+    The number of device definitions to retrieve from the end of the list.
+    Mutually exclusive with `first`.
+    """
+    last: Int
+    """
+    A cursor for pagination. Retrieve device definitions before this cursor.
+    """
+    before: String
+    """
+    Filter the device definitions based on specific criteria.
+    """
+    filterBy: DevicedefinitionFilter
+  ): DeviceDefinitionConnection!
+}
+
+"""
+Represents a Device Definition.
+"""
+type DeviceDefinition implements Node {
+  """
+  An opaque global identifier for this device definition.
+  """
+  id: ID!
+
+  """
+  Legacy ID for this device definition.
+  """
+  ksuid: String
```

Comment:
Can we name this something else, maybe `LegacyID`. KSUID doesn't describe what the variable does.